From: msvensson at mysql dot com
Date: December 22 2006 12:05am
Subject: bk commit into 5.0 tree (msvensson:1.2352) BUG#24148

Below is the list of changes that have just been committed into a local
5.0 repository of msvensson. When msvensson does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet@stripped, 2006-12-22 00:05:40+01:00, msvensson@stripped +1 -0
  Bug#24148 regression tests hang with SSL enabled
   - Don't call SSL_shutdown a second time

  vio/viossl.c@stripped, 2006-12-22 00:05:39+01:00, msvensson@stripped +9 -5
    SSL_shutdown might return 0 indicating that it should be called once
    again for a graceful shutdown. Since the socket are going to be closed
    anyway ther is no need for the second call.

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	msvensson
# Host:	pilot.mysql.com
# Root:	/home/msvensson/mysql/mysql-5.0-maint

--- 1.38/vio/viossl.c	2006-11-20 21:41:56 +01:00
+++ 1.39/vio/viossl.c	2006-12-22 00:05:39 +01:00
@@ -126,12 +126,16 @@
   {
     switch ((r= SSL_shutdown(ssl)))
     {
-    case 1: /* Shutdown successful */
+    case 1:
+      /* Shutdown successful */
+      break;
+    case 0:
+      /*
+        Shutdown not yet finished - since the socket is going to
+        be closed there is no need to call SSL_shutdown() a second
+        time to wait for the other side to respond
+      */
       break;
-    case 0: /* Shutdown not yet finished, call it again */
-      if ((r= SSL_shutdown(ssl) >= 0))
-        break;
-      /* Fallthrough */
     default: /* Shutdown failed */
       DBUG_PRINT("vio_error", ("SSL_shutdown() failed, error: %d",
                                SSL_get_error(ssl, r)));
